### DISCUSSION TITLE
Installation is hanging at 'Creating users table' on production server

### DISCUSSION THREAD

**Discussion by thomasbaine** [Laravel-Backpack/community-forum#967] [Q&A (Help)]
I'm running/accessing Backpack perfectly fine on my development machine (mySQL 8.1.0, Laravel 10.48.3, PHP 8.2.10), but after uploading it onto the production server (mySQL 8.0.35 for Linus, Laravel 10.48.4.0, PHP 8.3.1), I receive an Error 500 when I move to the login page (https:/xxxx/admin/login.

The backpack modules are:

backpack/basset: 1.3.0
backpack/crud: 6.7.11
backpack/generators: v4.0.5
backpack/permissionmanager: 7.2.0
backpack/pro: 2.1.13
backpack/theme-coreuiv4: 1.1.1

There are no errors posted in the apache2 or php-fpm logs, or in storage\logs\laravel. No error information appears in the browser's dev tools.

The public side of the site (a Vue PWA) - which is fuelled from the same database - is working perfectly.

I tried re-installing backpack on the production server:
	composer require backpack/crud
	php artisan backpack:install
	
the installation freezes at Step 1: Creating users table. The status stays as RUNNING forever.  As the table already existed in the database, I removed it and ran the installation again - but it still hung at the 'Creating users table' (sub)step.

I've checked my .env file and can't find any oversight/fault in it.

What am I missing/should do next?

Thanks/Bruce

**Comment by jcastroa87** [comment]
Hello @thomasbaine 

Please check directory permissions, create storage alias, and set the domain correctly in .env

If that does not work, please try the manual install as our [docs](https://backpackforlaravel.com/docs/6.x/crud-how-to#how-to-manually-install-backpack) say

Cheers

**Reply by pxpm** [reply]
I would just add: **dont run backpack install in production**. 

Backpack install should be a run-once and forget command. I think you are not the first person to make this "confusion" on what should be done when deploying. Did you get that from our documentation somewhere ? 

Here is my reply to a user with a similar problem: https://github.com/Laravel-Backpack/CRUD/issues/5496#issuecomment-2056226468

@jcastroa87 probably worth to add this to our documentation somewhere ?

**Reply by thomasbaine** [reply]
Not 'confused', more like 'panicked'.  I've deployed 3 sites incorporating Backpack without any problem - but when this one gave me the error 500 without any logged errors, I clutched at straws and tried the 'install in production' route.

**Reply by thomasbaine** [reply]
Thanks @jcastroa87 - but can you expand on creating a storage alias please.  I've checked the permissions of the folders in the app and they're all read/write - are these the directory permissions you referred to?  

Picking up from @pxpm's comment and @alonmoshi's comment within that thread, I switched APP_ENV to 'local' on the production side and installation was fine.  I've triple-checked the domain name (and changed the env setting back to 'production').

But I'm still getting the error 500 when I try to access BackPack at https://staging.xxxx.co.nz/admin/login.  Suggestions?

Thanks/Bruce

**Reply by pxpm** [reply]
@thomasbaine I don't think that's what me or Alan said there. Alan found the problem was probably a bad configured VHOST, and I totally said to not run the command when deploying. 

You should NOT run `backpack:install` in production/staging in any circumstance. 

Is you `APP_URL` in `.env` file `https://staging.xxx.co.nz` ? 

If you run `php artisan basset:check` in that server, what's the output ? 

Cheers

**Reply by thomasbaine** [reply]
Yes, I've checked the APP_URL many times and it is correct.

The basset:check output is:
   INFO  Checking Backpack Basset installation.

  Initializing basset check ............................................. DONE
  Checking cache storage ................................................ DONE
  Fetching a basset ..................................................... DONE

Should there be something else being reported?

Thanks/Bruce

**Reply by pxpm** [reply]
Hey @thomasbaine 
I think that should be fine from basset perspective. 

Any errors in your browser console ? 

If you do a `php artisan route:list` do you see the backpack registered routes ?

Running the following commands:
- `php artisan optimize:clear`
- `php artisan basset:clear`
- `php artisan basset:cache`
- `php artisan optimize`

Help in anything ? 

Cheers

**Comment by jcastroa87** [comment]
Due to not activity I will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.
